### PR TITLE
KOGITO-4635 Use tagLocalAndRemoteRepository to override tag if already exist

### PR DESCRIPTION
**Pull request by dupliaka**
### JIRA

Part of:
https://issues.redhat.com/browse/KOGITO-4635

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
</details>

### CODE REVIEW

**Comment by dupliaka** [comment]
Should be merged after this https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/146/files

**Review by radtriste — CHANGES_REQUESTED** [review]
Thx for the PR !
I was planning to update that as part of https://issues.redhat.com/browse/KOGITO-4635
I guess you can just add the issue number in title/description :)

**Comment by dupliaka** [comment]
> Thx for the PR !
> I was planning to update that as part of https://issues.redhat.com/browse/KOGITO-4635
> I guess you can just add the issue number in title/description :)

Thanks for that! Added issue number

File: Jenkinsfile.promote [hunk 1]
```diff
@@ -435,10 +435,9 @@ void mergeAndPush(String prLink) {
     mergeAndPush(prLink, getBuildBranch())
 }
 
-void tagLatest() {
+void tagLatest(String repo) {
```
Comment (radtriste) [inline Jenkinsfile.promote:438]: with latest change, I guess the parameter is not more needed.

Comment (dupliaka) [reply Jenkinsfile.promote]: Thanks you, I wish I had the same catchy eyes as you!

**Review by radtriste — APPROVED** [review]
lgtm

File: Jenkinsfile.promote (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile.promote]: always harder to see when it is your own code ;)

**Comment by dupliaka** [comment]
@radtriste could you please merge it as well? I do not have permissions

**Pull request merged**